proposals/function-pointers.md

```diff
@@ -253,83 +253,87 @@ conventions is very rich.
 the conversion to `object` can't be allowed, it can't be a member of a `class`, etc ... The C# design is to require
 `unsafe` for all pointer uses and hence this design follows that.
 
-Developers will still be capable of preventing a _safe_ wrapper on top of `func*` values the same way that they do 
+Developers will still be capable of preventing a _safe_ wrapper on top of `delegate*` values the same way that they do
 for normal pointer types today. Consider:
 
 ``` csharp
 unsafe struct Action {
-    func* void() _ptr;
+    delegate*<void> _ptr;
 
-    Action(func* void() ptr) => _ptr = ptr;
+    Action(delegate*<void> ptr) => _ptr = ptr;
     public void Invoke() => _ptr();
 }
 ```
 
 ### Using delegates
-Instead of using a new syntax element, `func*`, simply use existing `delegate` types with a `*` following the type:
+
+Instead of using a new syntax element, `delegate*`, simply use existing `delegate` types with a `*` following the type:
 
 ``` csharp
 Func<object, object, bool>* ptr = &object.ReferenceEquals;
 ```
 
-Handling calling convention can be done by annotating the `delegate` types with an attribute that specifies 
+Handling calling convention can be done by annotating the `delegate` types with an attribute that specifies
 a `CallingConvention` value. The lack of an attribute would signify the managed calling convention.
 
 Encoding this in IL is problematic. The underlying value needs to be represented as a pointer yet it also must:
 
-1. Have a unique type to allow for overloads with different function pointer types. 
+1. Have a unique type to allow for overloads with different function pointer types.
 1. Be equivalent for OHI purposes across assembly boundaries.
 
-The last point is particularly problematic. This mean that every assembly which uses `Func<int>*` must encode 
-an equivalent type in metadata even though `Func<int>*` is defined in an assembly though don't control. 
-Additionally any other type which is defined with the name `System.Func<T>` in an assembly that is not mscorlib 
+The last point is particularly problematic. This mean that every assembly which uses `Func<int>*` must encode
+an equivalent type in metadata even though `Func<int>*` is defined in an assembly though don't control.
+Additionally any other type which is defined with the name `System.Func<T>` in an assembly that is not mscorlib
 must be different than the version defined in mscorlib.
 
-One option that was explored was emitting such a pointer as `mod_req(Func<int>) void*`. This doesn't 
+One option that was explored was emitting such a pointer as `mod_req(Func<int>) void*`. This doesn't
 work though as a `mod_req` cannot bind to a `TypeSpec` and hence cannot target generic instantiations.
 
 ### Named function pointers
+
 The function pointer syntax can be cumbersome, particularly in complex cases like nested function pointers. Rather than
```

Comment:
>The [](start = 0, length = 3)

Please say something to make it clear that this is an alternative to, and not part of, the spec.

Reply:
This is bigger section titled `Considerations`, with a bunch of different future proposals.

---
In reply to: [344939001](https://github.com/dotnet/csharplang/pull/2923#discussion_r344939001) [](ancestors = 344939001)